modules/fedimint-ln-client/src/recurring/mod.rs

```diff
@@ -0,0 +1,525 @@
+            let await_any_invoice: BoxFuture<_> = if all_recurring_invoice_futures.is_empty() {
+                Box::pin(pending())
+            } else {
+                Box::pin(select_all(all_recurring_invoice_futures))
+            };
+
+            let (payment_code_idx, _payment_code, invoice_idx, invoice) = select! {
+                (ret, _, _) = await_any_invoice => match ret {
+                    Ok(ret) => ret,
+                    Err(_) => {
+                        continue;
+                    }
+                },
+                () = new_code_registered_future => {
+                    continue;
+                }
+            };
+
+            Self::process_recurring_payment_code_invoice(
+                &client,
+                payment_code_idx,
+                invoice_idx,
+                invoice,
+            )
+            .await;
+
+            // Just in case something goes wrong, we don't want to burn too much CPU
+            sleep(Duration::from_secs(1)).await;
+        }
+    }
+
+    async fn process_recurring_payment_code_invoice(
+        client: &ClientContext<Self>,
+        payment_code_idx: u64,
+        invoice_idx: u64,
+        invoice: lightning_invoice::Bolt11Invoice,
+    ) {
+        // TODO: validate invoice hash etc.
+        let mut dbtx = client.module_db().begin_transaction().await;
+        let old_payment_code_entry = dbtx
+            .get_value(&crate::db::RecurringPaymentCodeKey {
+                derivation_idx: payment_code_idx,
+            })
+            .await
+            .expect("We queried it, so it exists in our DB");
+
+        let new_payment_code_entry = RecurringPaymentCodeEntry {
+            last_derivation_index: invoice_idx,
+            ..old_payment_code_entry.clone()
+        };
+        dbtx.insert_entry(
+            &crate::db::RecurringPaymentCodeKey {
+                derivation_idx: payment_code_idx,
+            },
+            &new_payment_code_entry,
+        )
+        .await;
+
+        Self::create_recurring_receive_operation(
+            client,
+            &mut dbtx.to_ref_nc(),
+            &old_payment_code_entry,
+            invoice_idx,
+            invoice,
+        )
+        .await;
+
+        dbtx.commit_tx().await;
+    }
+
+    #[allow(clippy::pedantic)]
+    async fn create_recurring_receive_operation(
+        client: &ClientContext<Self>,
+        dbtx: &mut fedimint_core::db::DatabaseTransaction<'_>,
+        payment_code: &RecurringPaymentCodeEntry,
+        invoice_index: u64,
+        invoice: lightning_invoice::Bolt11Invoice,
+    ) {
+        // TODO: pipe secure secp context to here
+        let invoice_key =
+            tweak_user_secret_key(SECP256K1, payment_code.root_keypair, invoice_index);
+
+        // TODO: use proper operation id, what do we use for LN normally? Preimage I
+        // guess?
+        let operation_id = OperationId(*invoice.payment_hash().as_ref());
```

Comment:
We use the payment hash in LN normally, just like you do here https://github.com/fedimint/fedimint/blob/f82d08cdfbe30dd33069de7f2c660537a33acccb/modules/fedimint-ln-client/src/lib.rs#L965